.github/scripts/PR_reviewer.js

```diff
@@ -1,145 +1,216 @@
-		body = body.replaceAll(/\* (.+) (by .+) in (https:\/\/.+)(\n)*/g, '* [$1]($3) $2$4'); // Convert PR title to hyperlink
-		const summary = body.split(sectionTitle)?.[1]?.split('## Type of change')[0];
-		const review = body.split(reviewTitle)?.[1]; // Split review from PR description
-		body = `
-Reviewing [${title}](${link}) on \`${repoOwner}/${repoName}\`
-${sectionTitle}
-${summary.trim()}
-
-${reviewTitle}
-${review.trim()}
-`;
-		body = format(body);
-		body = body.replaceAll('\\r\\n', '');
-		console.info('body to send to Slack:', body);
-		
-		const { ok, error } = await slack.chat.postMessage({
-			channel: 'C075B3XH9AR', // #dev-github-security
-			text: body
-		})
-		if (error) {
-			console.error('Error notifying slack:', error);
-		}
-		return !!ok;
-	} catch (e) {
-		console.error('Error notifying slack:', e);
-		return false;
-	}
+  try {
+    const { sectionTitle, reviewTitle, reviewPR, format } = data;
+
+    const title = context.payload.pull_request.title;
+    const link = context.payload.pull_request.html_url;
+    const bodyPR = context.payload.pull_request.body;
+
+    // Determine priority dynamically using OpenAI
+    const priorityResponse = await openai.chat.completions.create({
+      messages: [{ role: 'user', content: getPriorityPrompt(reviewPR) }],
+      model: 'gpt-4o-mini',
+      temperature: 0.0,
+      max_tokens: 20,
+    });
+    const priority = priorityResponse?.choices[0].message.content.trim();
+    const priorityEmoji = getPriorityEmoji(priority);
+
+    // Create embedded hyperlink for Slack message
+    const embeddedLink = `<${link}|${title}>`;
+
+    // Format repository name in code style
+    const formattedRepoName = `\`${context.repo.owner}/${context.repo.repo}\``;
+
+    // Main Post Body with Priority and Emoji
+    const mainPostBody = `Reviewing ${embeddedLink} on ${formattedRepoName}\n*Priority: ${priority} ${priorityEmoji}*`;
+
+    // Thread Reply Body
+    let threadBody = bodyPR + reviewPR;
+    threadBody = threadBody.replace(/\n\s+-/g, '\n-'); // Remove whitespace between line break and bullet
+    threadBody = threadBody.replace(/\* (.+) (by .+) in (https:\/\/.+)(\n)*/g, '* [$1]($3) $2$4');
+
+    const summary = threadBody.split(sectionTitle)?.[1]?.split('## Type of change')[0]?.trim() || '';
+    const review = threadBody.split(reviewTitle)?.[1]?.trim() || '';
+
+    const formattedThreadReply = format(`${sectionTitle}\n${summary}\n\n${reviewTitle}\n${review}`);
+
+    // Send the Main Post to Slack
+    const { ok: postOk, ts, error: postError } = await slack.chat.postMessage({
+      channel: 'C07M93ERY6N', // #dele-test
+      text: mainPostBody,
+    });
+
+    if (!postOk) {
+      console.error(`Error posting main review to Slack: ${postError}`);
+      return false;
+    }
+
+    // Send the Thread Reply to Slack under the Main Post
+    const { ok: threadOk, error: threadError } = await slack.chat.postMessage({
+      channel: 'C07M93ERY6N', // #dele-test
```

Comment:
it would be better to create as a variable as you are re-using it
```suggestion
      channel: 'C075B3XH9AR', // #dev-github-security
```

Reply:
yes sure